sdk/core/Azure.Core/src/ExponentialDelayStrategy.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public class ExponentialDelayStrategy : DelayStrategy
+    {
+        private readonly TimeSpan _delay;
+        private readonly TimeSpan _maxDelay;
+        private readonly Random _random = new ThreadSafeRandom();
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="delay"></param>
+        /// <param name="maxDelay"></param>
+        public ExponentialDelayStrategy(TimeSpan? delay = default, TimeSpan? maxDelay = default)
+        {
+            // use same defaults as RetryOptions
+            _delay = delay ?? TimeSpan.FromSeconds(0.8);
+            _maxDelay = maxDelay ?? TimeSpan.FromMinutes(1);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="response"></param>
+        /// <param name="attempt"></param>
+        /// <param name="delayHint"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public override TimeSpan GetNextDelay(Response? response, int attempt, TimeSpan? delayHint)
+        {
+            return
+                Max(
+                    delayHint ?? TimeSpan.Zero,
+                    Max(
+                        GetServerDelay(response),
```

Comment:
I would consider flipping this back around.  If done this way we introduce a scenario where customers could write their own delay strategy and bypass the RetryAfter headers.  The RetryAfter headers are there to protect azure services from being spammed more than to help the customer get the fastest response.

Originally we had a RetryAfterDelayStrategy which took a fallback strategy incase it didn't exist or the customer opted into waiting longer.  This could still exist and be internal which would allow the customer to create their own strategy but not introduce a scenario would the customer is spamming our services and causing issues.

Reply:
This will depend on whether we want to allow customers to bypass the retry-after headers or not. The approach taken here is to provide an API that implementors can use, `GetServerDelay`, if they want to override GetNextDelay but still respect the server delay. If we don't want to allow this level of control, then I agree with your proposed approach.

@KrzysztofCwalina, @tg-msft what do you think about this issue? Should customers be allowed to bypass the retry-after header, or should they be limited to Max(Retry-After, CustomDelay)?

Reply:
@m-nash provided more context around the LRO use case. The impact of a customer ignoring the server delay in a custom strategy is larger for LROs because we expect to poll the server multiple times for each LRO. For retries, it is a bit different because retry-after header would only come into play for error scenarios.

Reply:
Pinging @KrzysztofCwalina @tg-msft, what do you think the ideal behavior should be for the retry case? Should we allow users to ignore the server delay or would it be better to always wrap their delay strategy so that we use the max returned by their strategy and the retry header.

Reply:
I think we cannot prevent customers from ignoring the header. They can write their own client after all, or they can change the retry policy to only do one try and then "retry" in the client calling code. 

But I do agree that it would be good to design the extensibility APIs such that the user has to explicitly ignore the header as opposed to just not knowing that they should honor it. Maybe we can add "suggestedRetryAfter" parameter to the extensibility APIs.

Reply:
Just to clarify - @m-nash is suggesting that we don't allow customers to ignore the header for LRO scenarios, because the penalty of doing so would be quite severe. Are you alright with us diverging between LRO and retries?

Reply:
One difference between general retries and LROs is that Retries usually have a finite retry limit. I think that makes ignoring the header less impactful to the service. That said, I agree that the default should be to respect to header with some way to explicitly override it.

Reply:
Yes, also retries only occur in error scenarios, where trying again is expected for LROs. The default is to respect the header, but we are discussing whether it should be possible to override the header for LROs and for retries. @m-nash has stated that it should not be overrideable for LROs.

Reply:
Do we have two different policies for LROS and not LROS?

Reply:
We don't exactly have policies for LROs, but we do have delay strategies (currently internal). The delay strategies are applied outside of the pipeline, whereas retry policy is within the policy. So an LRO would use the configured retry policy for individual requests, but between requests it would use the LRO delay strategy.

Reply:
There is a big difference between retry and lro-polling in that during a retry scenario the response is not expected and in most cases can / should be remedied quickly.  In an lro-polling scenario the response is expected and the retry-after protects not only azure services but it protects the customer from accidentally locking their subscription by hitting their api rate limit.  On top of this there is no expectation that this should get "fixed quickly".  For one, nothing is broken to fix, and for two some of these LROs take a very long time to finish.

My suggestion is that when you pass in a DelayStrategy to WaitForCompletion that this is treated as the fallback strategy to the RetryAfterDelayStrategy that exists today.  If someone writes their own rest client to spam themselves into api rate limiting thats on them, but our client shouldn't make this the default path.  Even adding a `useSuggestedRetryAfter` won't necessarily be understood by customers and I can't come up with a single valid reason for anyone to need to do that.

Reply:
> If someone writes their own rest client to spam themselves into api rate limiting thats on them, but our client shouldn't make this the default path.

This wouldn't be the default path. In order for customers to do this, they'd need to create their own custom delay strategy, i.e. not use one of the out-of-the-box ones for Exponential, Fixed, or Sequential, and in their custom delay strategy they would need to override GetNextDelay and choose to ignore the serverDelayHint.

Reply:
I think the point still stands that they will have no idea what it is the are opting into or out of.

What is the legitimate scenario where a customer needs the ability to spam the service instead of waiting the suggested amount of time?

Reply:
> What is the legitimate scenario where a customer needs the ability to spam the service instead of waiting the suggested amount of time?

Bugs in the service that result in bogus retry-after header values is one I can imagine.

Reply:
@tg-msft had a suggestion that we introduce GetNextDelayCore as abstract which would not take either hint. We could then have GetNextDelay as virtual which has a default implementation that uses the max of the hints. If customers really want to, they can override this, but it makes it that much more hidden.